fish-pepper/run-java-sh/fp-files/run-java.sh

```diff
@@ -129,7 +129,7 @@ calc() {
     function round(x) {
       return int(x + 0.5)
     }
-    {printf "%d\n",'"${formula}"'}
+    {print '"${formula}"'}
```

Comment:
I think we need to review the places that call the `calc` function and that may require integer output.

Reply:
At first glance, there is only the `ci_compiler_count` method that does not explicitly round / ceil the result and that may lead to an invalid JVM option value.

Reply:
Yet the tests pass :D

Reply:
maybe we can add ad int(...) into log2 ? So that we guarantee that every function already returns an int ? (which implicitly happened anyway because of the %d at the end ...)

Reply:
See latest push ...

Reply:
The original OpenJDK algorithm uses int as intermediate results so that works, in the sense the script mimics the exact same values as the JVM. That being said, I think that would mean putting the `int` as the final stage, e.g.:

    {print 'int("${formula}")'}

as opposed to each individual functions.

Reply:
> Yet the tests pass :D

not sure whether we test ci_compiler_count tbh

@astefanutti BTW, have you seen the latest recommendation from Andrig Miller for the middleware images ? They recommend to always set -Xms to 1/4 of -Xmx. Not sure why though, but seemed to be a quite generic recommendation.

Reply:
@rhuss yes I've seen it. Actually we already did that with aed74685c7253811653a6e01a8cffe67d2907e67. I wanted to do it generically, though you recommended we keep the 50% default for backward compatibility, so we are applying it when available memory is below 300MB.

Reply:
> not sure whether we test ci_compiler_count tbh

We added our tests campaign after the introduction of `ci_compiler_count` and we're checking the script output without running the JVM so that explain while the tests are green. I should add an explicit test on the `ci_compiler_count` value!

Reply:
> Actually we already did that with aed7468

Hmm, I don't see where we set `-Xms` as 25% of `-Xmx` (only when we set JAVA_INIT_MEM_RATIO), so I wonder whether we should add this automatically when JAVA_MAX_MEM_RATIO is set ?

Reply:
Ah I misread the point. From the presentation:
> To account for all memory, for a 1 GB memory container, max heap should never be larger than 512MB, and the min should always be 25% of the max when using the above [JVM GC options].

Not sure what that exactly means and what's the rational...